MiniApp/Classes/ui/MiniAppViewController.swift

```diff
@@ -0,0 +1,188 @@
+import Foundation
+import UIKit
+
+public protocol MiniAppUIDelegate: class {
+    func onError(error: Error)
+    func onSuccess()
+    func onForward()
+    func onBackward()
+    func onClose()
+}
+public extension MiniAppUIDelegate {
+    func onError(error: Error) {}
+    func onSuccess() {}
+    func onForward() {}
+    func onBackward() {}
+}
+
+public class MiniAppViewController: UIViewController {
+
+    let appId: String
+    var config: MiniAppSdkConfig?
+    var queryParams: String?
+    var state: ViewState = .loading {
+        didSet { update() }
+    }
+
+    weak var messageDelegate: MiniAppMessageDelegate?
+    weak var navDelegate: MiniAppNavigationDelegate?
```

Comment:
There are some additional concerns about this we need to think about:

- We need to let the Host App provide their own `MiniAppNavigationDelegate` so that they can override certain links (i.e. deep links) or if there are certain links they want to handle differently.
- We need to also provide a default WebView for displaying links to external pages. The SDK already has a default WebView built-in, but it doesn't provide the full functionality because it needs access to the ViewController. Now, we have access to the ViewController here, so we can provide the full functionality (see the Example app code):
  - The Mini App has the capability to receive parameters back from the external page which was opened in a secondary WebView (and the external page has the capability to close the secondary WebView). This is done by handling a specific `miniapp://` scheme from the secondary WebView.

So basically I think we need to have two `MiniAppNavigationDelegate`: one which is provided by the Host App (and we check this one first to see if the Host App handles the link) and one which is a default in this ViewController.

Reply:
@corycaywood which links exactly should I handle per default implementation?